profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_dgt_connector/tmgmt_dgt_connector.inc

```diff
@@ -0,0 +1,298 @@
+  // language combination add add the relevant job items to it.
+  foreach ($target_languages as $target_language) {
+    foreach ($job_items_by_source_language as $source_language => $job_items) {
+      // Skip in case the source language is the same as the target language.
+      if ($source_language == $target_language) {
+        continue;
+      }
+
+      if (!empty($reference)) {
+        // Poetry: We need the reference to identify related jobs.
+        $job = tmgmt_job_create($source_language, $target_language, $user->uid, $reference);
+      }
+      else {
+        $job = tmgmt_job_create($source_language, $target_language, $user->uid);
+      }
+
+      $job_empty = TRUE;
+      /** @var TMGMTJobItem $job_item */
+      foreach ($job_items as $id => $job_item) {
+        try {
+          // As the same item might be added to multiple jobs, we need to
+          // re-create them and delete the old ones, after removing them from
+          // the cart.
+          $job->addItem($job_item->plugin, $job_item->item_type, $job_item->item_id);
+          $remove_job_item_ids[$job_item->tjiid] = $job_item->tjiid;
+          $job_empty = FALSE;
+        }
+        catch (Exception $e) {
+          // If an item fails for one target language, then it is also going
+          // to fail for others, so remove it from the array.
+          unset($job_items_by_source_language[$source_language][$id]);
+          drupal_set_message($e->getMessage(), 'error');
+        }
+      }
+
+      if (!$job_empty) {
+        $jobs[] = $job;
+
+        // Set reference to be based on main job ID.
+        if (empty($reference)) {
+          $reference = array('reference' => 'SUB_' . $job->tjid);
+        }
+      }
+
+    }
+  }
+
+  // Remove job items from the cart.
+  if ($remove_job_item_ids) {
+    tmgmt_ui_cart_get()->removeJobItems($remove_job_item_ids);
+    entity_delete_multiple('tmgmt_job_item', $remove_job_item_ids);
+  }
+
+  // Start the checkout process if any jobs were created.
+  if ($jobs) {
+    tmgmt_ui_job_checkout_and_redirect($form_state, $jobs);
+  }
+  else {
+    drupal_set_message(t('It is not possible to create a translation job from the selection you made.'), 'error');
+  }
+
+  // Store chosen translation request languages for further processing.
+  _tmgmt_poetry_workbench_store_request_languages($target_languages);
+
+  // Empty redirection queue, Poetry doesn't work with that.
+  tmgmt_ui_redirect_queue_set(array());
+}
+
+/**
+ * Custom function to manage permissions of the tmgmt_dgt_connector translator.
+ *
+ * @param string $op
+ *   Which operation is performed.
+ * @param array $translator
+ *   The translator the operation is performed on.
+ * @param array $account
+ *   The account which is performing the update.
+ *
+ * @return bool
+ *   Boolean
+ */
+function _tmgmt_dgt_connector_translator_access($op, array $translator, array $account) {
+  if (!empty($translator) && $translator->plugin == 'tmgmt_dgt_connector') {
```

Comment:
`$translator` is supposed to be an array.

How could you can treat it as an object ?

If you never had any issue with this, it means that the function has never been triggered maybe...